packages/rpc-core/src/rpc-methods/getSignaturesForAddress.ts

```diff
@@ -0,0 +1,44 @@
+import { Base58EncodedAddress } from '@solana/keys';
+import { Commitment, RpcResponse, Slot, TransactionSignature, U64UnsafeBeyond2Pow53Minus1 } from './common';
+
+type GetSignaturesForAddressTransaction = RpcResponse<{
+    /** transaction signature as base-58 encoded string */
+    signature: TransactionSignature;
+    /** The slot that contains the block with the transaction */
+    slot: Slot;
+    /** Error if transaction failed, null if transaction succeeded. */
+    err: object | null;
+    /** Memo associated with the transaction, null if no memo is present */
+    memo: string | null;
+    /** estimated production time, as Unix timestamp (seconds since the Unix epoch) of when transaction was processed. null if not available. */
+    blockTime: U64UnsafeBeyond2Pow53Minus1 | null;
+    /** The transaction's cluster confirmation status */
+    confirmationStatus: Commitment | null;
+}>;
```

Comment:
Oh, actually, this is totally worth making a union type for, because if someone asserts on `err` being `null`, Typescript will narrow it to the right type.

https://www.typescriptlang.org/play?ssl=16&ssc=1&pln=16&pc=2#code/C4TwDgpgBAKgTgQwHYGcEGNgEsD2SCiccOcUAvFAN4BQUUERAXFDgEYBWEm1AvtdaEixEqDNjzkqtekyhIArgBtF0gLYRVOZimBwsSAOZQAPnKWKA3L37o8OqMAAezeMjSZcSE8LdjPhYlIKSh4oBBQwpBAraiwAMygACicAOgYgigVlAEopOjpU9U0LKAB6UqgAeQBpXnpFFGgafIdHFKKcEvKofAAlXt4gA

Reply:
Do you know if `memo` is guaranteed not to be present if there's an error? It looks like everything else is still there:

```json
        {
            "blockTime": 1671619698,
            "confirmationStatus": "finalized",
            "err": {
                "InstructionError": [
                    0,
                    "IllegalOwner"
                ]
            },
            "memo": null,
            "signature": "66CXkJDxKr52jxxGec4k3cK4h5R5edWZXsfxH9w3CcbtwnrAK4awK8S1LiSE7h768M4rvH332TAs3MQZ4sGwvSGF",
            "slot": 168164198
        },
```

But I don't know of any examples of failing transactions with memo instructions to check

Reply:
Ah. Point taken. All the properties are there all the time then. Got it.

```
curl https://api.mainnet-beta.solana.com -X POST -H "Content-Type: application/json" -d '
  {
    "jsonrpc": "2.0",
    "id": 1,
    "method": "getSignaturesForAddress",
    "params": [
      "MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr",
      {
        "before":"4aBLTpNXCtugZUfyNenEoAA7tjHnJ8j4AJ8jpDZQotCvbR2A7AXX3KfwfKkfXQsp83yYNQdW1TQCyrh6njK1ZPtd","limit":1
      }
    ]
  }
'
{"jsonrpc":"2.0","result":[{"blockTime":1684351582,"confirmationStatus":"finalized","err":{"InstructionError":[2,{"Custom":6001}]},"memo":"[27] \"Sending tx from 0 1.6.3 1\"","signature":"3ddcFK6Kf8y3r8TMF1PJV8V2Cp6TkyWV2Y3odS47JW9zWPfb8NMFxVnG3GSoGzPJ6JCGdhJmurwQMnpL2LoBdtCT","slot":194464790}],"id":1}
```

Contains the memo, even for a failed tx.